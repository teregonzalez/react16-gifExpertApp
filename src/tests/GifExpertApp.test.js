import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';
describe('Test en GifExpertApp', () => { 
    test('Debe mostrarse correctamente', () =>{

        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot()

    });
    test('Debe de mostrar una lista de categorias', () => { 
        
        const categories = ['HunterxHunter', 'Full Metal Alchemist'];

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe( categories.length );
    })
 });