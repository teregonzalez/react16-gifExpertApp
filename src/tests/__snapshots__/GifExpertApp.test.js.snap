// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="HunterxHunter"
      key="HunterxHunter"
    />
    <GifGrid
      category="Full Metal Alchemist"
      key="Full Metal Alchemist"
    />
  </ol>
</Fragment>
`;

exports[`Test en GifExpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
